File: src/main/java/io/github/funofprograming/context/impl/AbstractApplicationContextMergeStrategy.java
//package io.github.funofprograming.context.impl;
//
//import java.util.Objects;
//import java.util.Set;
//
//import io.github.funofprograming.context.Key;
//import io.github.funofprograming.context.ApplicationContextMergeStrategy;
//
///**
// * Abstract implementation of {@linkplain ApplicationContextMergeStrategy}
// *
// * @author Akshay Jain
// *
// */
//public abstract class AbstractApplicationContextMergeStrategy implements ApplicationContextMergeStrategy {
//
//
//    protected final Set<Key<?>> keys;
//
//    /**
//     * Initialize with a set of keys for whom this strategy is to be applied. If null set passed then strategy is applicable for all keys
//     *
//     * @param keys
//     */
//    protected AbstractApplicationContextMergeStrategy(Set<Key<?>> keys)
//    {
//        this.keys = keys;
//    }
//
//    protected boolean keySetAvailable()
//    {
//        return Objects.nonNull(keys);
//    }
//
//    protected boolean keyAvailable(Key<?> key)
//    {
//        return keySetAvailable()?keys.contains(key):false;
//    }
//
//}

File: src/main/kotlin/io/github/funofprograming/context/impl/AbstractApplicationContextMergeStrategy.kt
package io.github.funofprograming.context.impl

import io.github.funofprograming.context.ApplicationContextMergeStrategy
import io.github.funofprograming.context.Key

/**
 * Abstract implementation of {@linkplain ApplicationContextMergeStrategy}
 *
 * Initialize with a set of keys for whom this strategy is to be applied. If null set passed then strategy is applicable for all keys
 * @param keys
 *
 *  @author Akshay Jain
 *
 */
abstract class AbstractApplicationContextMergeStrategy(protected var keys: Set<Key<*>>? = null) : ApplicationContextMergeStrategy {

    protected fun keySetAvailable(): Boolean {
        return keys.isNullOrEmpty()
    }

    protected fun keyAvailable(key: Key<*>): Boolean {
        return keys?.contains(key) ?: false
    }
}